gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -143,14 +145,25 @@ public GsonBuilder() {
   }
 
   /**
-   * Configures Gson to enable versioning support.
+   * Configures Gson to enable versioning support. Versioning support works based on the
+   * annotation types {@link Since} and {@link Until}. It allows including or excluding fields
+   * and classes based on the specified version. See the documentation of these annotation
+   * types for more information.
    *
-   * @param ignoreVersionsAfter any field or type marked with a version higher than this value
-   * are ignored during serialization or deserialization.
+   * <p>By default versioning support is disabled and usage of {@code @Since} and {@code @Until}
+   * has no effect.
+   *
+   * @param version the version number to use.
    * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
+   * @throws IllegalArgumentException if the version number is NaN or negative
+   * @see Since
+   * @see Until
    */
-  public GsonBuilder setVersion(double ignoreVersionsAfter) {
-    excluder = excluder.withVersion(ignoreVersionsAfter);
+  public GsonBuilder setVersion(double version) {
+    if (Double.isNaN(version) || version < 0.0) {
```

Comment:
Have added this `< 0.0` check because negative version numbers are probably rather uncommon and this could cause collisions with the undocumented value `-1.0` representing no version:
https://github.com/google/gson/blob/28609089faa747f2ad5730281c14093ab40d6fda/gson/src/main/java/com/google/gson/internal/Excluder.java#L51